### PR TITLE
Naming manipulation tools to facilitate importing model weights from …

### LINKED ISSUE #42: More control over var/module namespace.
I got my first 'hello world' model experiment working w/ Objax. I adapted my PyTorch EfficientNet impl. Overall pretty smooth, currently wrapping Conv2d so I can get the padding I want.

One thing that stuck out after inspecting the model, the var namespace is a mess. An aspect of modelling that I value highly is the ability to have sensible checkpoint/var maps to work with. I often end up dealing with conversions between frameworks, exports for mobile or embedded targets and having your vars (parameters) sensibly named, and often being able to control those names in the originating framework is important. 

Any thoughts on improving this? The current name/scoping mechanism forces the inclusion of the Module class names, is that necessary? Shouldn't attr names through the tree be enough for uniqueness?

Also, there is no ability to specify names for modules in sequential containers. I use this quite often for frameworks that have it. Sometimes I don't care much (long list of block repeats, 0..n is fine), but for finer grained blocks I like to know what conv is what by looking at the var names. '0.b, o.w' etc isn't very useful.

I'll post an example of the var keys below, and comparison point for pytorch.

### PR DESCRIPTION
**Pull request by david-berthelot**
…other frameworks and to remap weights between models.

- VarCollection has a new rename method to rename its variables.
- load_var_collection has new option to rename variables from the data read from the file.
- Renamer allows to take a dictionary of strings {source:target}, or a tuple (regex, replacement) or a function (str -> str).

Resolves #42

### CODE REVIEW

**Comment by AlexeyKurakin** [comment]
Looks good.
Could you add test for `load_var_collection` with renamer?

**Review by AlexeyKurakin — APPROVED** [review]
